src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -159,19 +147,24 @@ private OverdueFineCalculatorService(Clients clients) {
     }
 
     return CompletableFuture.completedFuture(succeeded(
-      new CalculationParameters(loan, null, null, null)))
+      new CalculationParameters(loan)))
       .thenCompose(r -> r.after(this::lookupItemRelatedRecords))
       .thenCompose(r -> r.after(this::lookupFeeFineOwner))
       .thenCompose(r -> r.after(this::lookupFeeFine))
-      .thenCompose(r -> r.after(params -> {
-        if (params.item == null || params.feeFineOwner == null || params.feeFine == null) {
-          return failure();
-        }
+      .thenCompose(r -> r.after(params -> createAccount(fineAmount, params)));
+  }
 
-        AccountStorageRepresentation account = new AccountStorageRepresentation(params.loan,
-          params.item, params.feeFineOwner, params.feeFine, fineAmount);
-        return accountRepository.create(account);
-      }));
+  private CompletableFuture<Result<Account>> createAccount(Double fineAmount,
+    CalculationParameters params) {
+
+    if (params.isIncomplete()) {
+      return failure();
```

Comment:
OverdueFineCalculatorServiceTest.java should have a unit tests for this failure. ItemLastCheckInTests.checkedInItemWithLoanShouldHaveLastCheckedInFields() executes the line but is not designed to test this.

Reply:
Done.